ui/lib/CreateStream.tsx

```diff
@@ -25,32 +26,67 @@ export interface CreateStreamProps {
 }
 
 const CreateStream = (props: CreateStreamProps) => {
+  const [balance, setBalance] = useState<number | null>(null);
+  const [tokenBalance, setTokenBalance] = useState<number | null>(null);
+  const [gasPrice, setGasPrice] = useState<number | null>(null);
+  const { data, isError, isLoading } = useFeeData();
   const { address } = useAccount();
+
+  // Set balances for native and stream tokens
+  useEffect(() => {
+    if (address) {
+      (async () => {
+        // Fetch and set native balance as integer
+        const balanceData = await fetchBalance({ address });
+        if (balanceData && balanceData.formatted !== undefined) {
+          setBalance(Number(balanceData.formatted));
+        }
+        // Fetch and set token balance as integer
+        const tokenBalanceData = await fetchBalance({
+          address,
+          token: props.tokenAddress,
+        });
+        if (tokenBalanceData && tokenBalanceData.formatted !== undefined) {
+          setTokenBalance(Number(tokenBalanceData.formatted));
+        }
+      })();
+    }
+  }, [address]);
```

Comment:
Just double checking my understanding of react, this will trigger the `useEffect` on every update of `address`?

Reply:
Yes this is the dependency array; it runs whenever there is a change in the address. Otherwise you'd be able to load the page then change your metamask account.